test/interpret.jl

```diff
@@ -415,7 +415,7 @@ end
 
     function g_gf()
         eval(:(z = 2))
-        return z
+        return @invokelatest(@__MODULE__().z)
```

Comment:
This test should either change like you suggest, switch to `@test_throws`, or go away. Currently it's testing that JuliaInterpreter is broken wrt world age, xref #666

FYI the `teh/worldage` branch is a start on proper support for world age, but it's on hold for now while I work on other package updates for 1.12.